iota-client/tests/node_api.rs

```diff
@@ -73,7 +73,7 @@ async fn test_post_message_with_indexation() {
 
     let r = client
         .send()
-        .indexation("Hello".to_string())
+        .with_index("Hello")
```

Comment:
How do you think we change all the name `index` to be indexation, so it is aligned with the full name described in the spec?

Reply:
After thinking twice with_index is better, because indexation has `index` and `data`, which are defined in the bee crates

Reply:
But good point about the spec, I already updated the name in https://github.com/iotaledger/iota.rs/pull/231, but the naming is probably not consistent everywhere yet